packages/@aws-cdk/aws-servicecatalogappregistry/test/application.test.ts

```diff
@@ -128,5 +129,82 @@ describe('Application', () => {
       });
     }).toThrow(/Invalid application name for resource/);
   });
-});
 
+  describe('Assocations on applications', () => {
+    let application: appreg.Application;
+
+    beforeEach(() => {
+      application = new appreg.Application(stack, 'MyApplication', {
+        applicationName: 'MyApplication',
+      });
+    });
+
+    test('associate attribute group', () => {
+      const attributeGroup = new appreg.AttributeGroup(stack, 'AttributeGroup', {
+        attributeGroupName: 'AttributeGroupName',
+        attributes: {},
+      });
+
+      application.associateAttributeGroup(attributeGroup);
+
+      expect(stack).toHaveResourceLike('AWS::ServiceCatalogAppRegistry::AttributeGroupAssociation', {
+        Application: { 'Fn::GetAtt': ['MyApplication5C63EC1D', 'Id'] },
+        AttributeGroup: { 'Fn::GetAtt': ['AttributeGroup409C6335', 'Id'] },
+      });
+    }),
+
+    test('duplicate attribute group association are idempotent', () => {
+      const attributeGroup = new appreg.AttributeGroup(stack, 'AttributeGroup', {
+        attributeGroupName: 'attributeGroupName',
+        attributes: { key: 'value' },
+      });
+
+      // If these were not idempotent, the second call would produce an error for duplicate construct ID.
+      application.associateAttributeGroup(attributeGroup);
+      application.associateAttributeGroup(attributeGroup);
```

Comment:
Maybe test idempotency in the below test, that has an assertion on the number of `AWS::ServiceCatalogAppRegistry::AttributeGroupAssociation` services?

Tests without assertions are not great.

Reply:
done, also put in the resource association one as well